### PR TITLE
feat(MSI bundler) Basic functionality. 

### PR DESCRIPTION
**Pull request by tensor-programming**
> **What kind of change does this PR introduce?** (check at least one)
> 
* [x]  Bugfix
* [x]  Feature
* [ ]  New Binding Issue #___
* [ ]  Code style update
* [x]  Refactor
* [ ]  Build-related changes
* [ ]  Other, please describe:

**Does this PR introduce a breaking change?** (check one)
 
 * [ ]  Yes. Issue #___
 * [ ]  No
 
 **The PR fulfills these requirements:**
 
 * [x ]  It's submitted to the `dev` branch and _not_ the `master` branch
 * [ ]  When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
 * [ ]  Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/proton/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

 If adding a **new feature**, the PR's description includes:

* [ ]  A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

 **Other information:**

Added features allow for a minimal MSI bundler to work on Windows x64 systems.  Currently, only generates a simple wrapper around the produced Rust exe file.  No support for logos or custom paths; however, these features could be added fairly rapidly.  A few items were also added to allow for cross compilation in the future.

### CODE REVIEW

File: tools/rust/cargo-proton-bundle/src/bundle/msi_bundle.rs [hunk 1]
```diff
@@ -1,36 +1,12 @@
+#[warn(dead_code)]
```
Comment (nothingismagick) [inline tools/rust/cargo-proton-bundle/src/bundle/msi_bundle.rs]: that's nice

**Comment by lucasfernog** [comment]
Unbeliavable how you Tensor did this with limited time.

File: tools/rust/cargo-proton-bundle/src/bundle/msi_bundle.rs (hunk 1 above)
Comment (tensor-programming) [reply tools/rust/cargo-proton-bundle/src/bundle/msi_bundle.rs]: Yeah, it can be removed but I put it in there for my own sanity.

**Comment by nothingismagick** [comment]
Do we need to do renaming here too?

**Comment by lucasfernog** [comment]
I think the only thing that might need renaming here is the bundler dir name (from cargo-proton-bundle to cargo-tauri-bundle).

**Comment by nothingismagick** [comment]
I'm on mobile at the moment, so its hard for me to check.

**Pull request merged**